torchx/test/torchx/backend_test.exs

```diff
@@ -388,4 +388,25 @@ defmodule Torchx.BackendTest do
                    end
     end
   end
+
+  describe "Nx.LinAlg.lu" do
+    test "property" do
+      for _ <- 1..20 do
+        a = Nx.random_uniform({3, 3})
+        {p, l, u} = Nx.LinAlg.lu(a)
+
+        a_reconstructed = p |> Nx.dot(l) |> Nx.dot(u)
+
+        assert_all_close(a, a_reconstructed)
+      end
+    end
```

Comment:
This test shows that the implementation works, even though the {p, l, u} matrices aren't the same as what binary backend would return.

Specifically:

```elixir
# For Torchx
{p, l, u} = Nx.LinAlg.lu(Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]]))  
{#Nx.Tensor<
   s64[3][3]
   Torchx.Backend(cpu)
   [
     [0, 1, 0],
     [0, 0, 1],
     [1, 0, 0]
   ]
>, #Nx.Tensor<
   f32[3][3]
   Torchx.Backend(cpu)
   [
     [1.0, 0.0, 0.0],
     [0.1428571492433548, 1.0, 0.0],
     [0.5714285969734192, 0.4999997913837433, 1.0]
   ]
>, #Nx.Tensor<
   f32[3][3]
   Torchx.Backend(cpu)
   [
     [7.0, 8.0, 9.0],
     [0.0, 0.8571428060531616, 1.7142856121063232],
     [0.0, 0.0, 1.7881390590446244e-7]
   ]
>}

# For BinaryBackend
{p, l, u} = Nx.LinAlg.lu(Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]], backend: Nx.BinaryBackend))
{#Nx.Tensor<
   s64[3][3]
   [
     [0, 0, 1],
     [0, 1, 0],
     [1, 0, 0]
   ]
>, #Nx.Tensor<
   f32[3][3]
   [
     [1.0, 0.0, 0.0],
     [0.5714285969734192, 1.0, 0.0],
     [0.1428571492433548, 2.0, 1.0]
   ]
>, #Nx.Tensor<
   f32[3][3]
   [
     [7.0, 8.0, 9.0],
     [0.0, 0.4285714328289032, 0.8571428656578064],
     [0.0, 0.0, 0.0]
   ]
>}
p |> Nx.dot(l) |> Nx.dot(u)                                                                      
#Nx.Tensor<
  f32[3][3]
  [
    [1.0, 2.0, 3.0],
    [4.0, 5.0, 6.0],
    [7.0, 8.0, 9.0]
  ]
>
```